core_lib/src/interface/editor.h

```diff
@@ -189,6 +187,16 @@ class Editor : public QObject
      * @return The newly created keyframe or `nullptr` if the layer is not visible
      */
     KeyFrame* addNewKey();
+        /**
+     * Attempts to create a new keyframe at the given position and layer. If a keyframe already exists at the position,
+     * the new one will instead be created on the first empty frame that follows. If the layer is not visible, a warning
+     * dialog will be shown to the user and no new keyframe is created.
+     * @param layerNumber The number of an existing layer on which the keyframe is to be created
+     * @param frameIndex The desired position of the new keyframe
+     * @return The newly created keyframe or `nullptr` if the layer is not visible
+     * @see Editor::addNewKey()
+     */
+    KeyFrame* addKeyFrame(int layerNumber, int frameIndex);
```

Comment:
This function wasn't meant to be public, that's why it seems a bit awkward and needs such a long comment to explain what happens under the hood. We can improve it later on in another PR.

Reply:
Ideally Editor shouldn't have these methods at all, we should utilize the LayerManager instead. I agree that it shouldn't be exposed, I did it because there was no alternative without making additional changes.